ecosystem/typescript/sdk/src/token_client.ts

```diff
@@ -98,34 +107,30 @@ export class TokenClient {
     property_values: Array<string> = [],
     property_types: Array<string> = [],
   ): Promise<Gen.HexEncodedBytes> {
-    const payload: Gen.TransactionPayload = {
-      type: "script_function_payload",
-      function: {
-        module: {
-          address: "0x3",
-          name: "token",
-        },
-        name: "create_token_script",
-      },
-      type_arguments: [],
-      arguments: [
-        Buffer.from(collectionName).toString("hex"),
-        Buffer.from(name).toString("hex"),
-        Buffer.from(description).toString("hex"),
-        supply.toString(),
-        NUMBER_MAX.toString(),
-        Buffer.from(uri).toString("hex"),
-        royalty_payee_address.toString(),
-        royalty_points_denominator.toString(),
-        royalty_points_numerator.toString(),
-        [false, false, false, false, false],
-        property_keys.map((key) => Buffer.from(key).toString("hex")),
-        property_values.map((val) => Buffer.from(val).toString("hex")),
-        property_types.map((typ) => Buffer.from(typ).toString("hex")),
-      ],
-    };
-    const transactionHash = await this.submitTransactionHelper(account, payload);
-    return transactionHash;
+    const payload = new TxnBuilderTypes.TransactionPayloadScriptFunction(
+      TxnBuilderTypes.ScriptFunction.natural(
+        "0x3::token",
+        "create_token_script",
+        [],
+        [
+          BCS.bcsSerializeStr(collectionName),
+          BCS.bcsSerializeStr(name),
+          BCS.bcsSerializeStr(description),
+          BCS.bcsSerializeUint64(supply),
+          BCS.bcsSerializeUint64(MAX_U64_BIG_INT),
```

Comment:
similar as above we could make 
`async createToken(
    account: AptosAccount,
    collectionName: string,
    name: string,
    description: string,
    supply: number,
    uri: string,
    max: bigint = MAX_U64_BIG_INT,
    royalty_payee_address: MaybeHexString = account.address(),
    royalty_points_denominator: number = 0,
    royalty_points_numerator: number = 0,
    property_keys: Array<string> = [],
    property_values: Array<string> = [],
    property_types: Array<string> = [],
  )`
and use max to replace `BCS.bcsSerializeUint64(MAX_U64_BIG_INT)` here